doc/cabal-project-description-file.rst

```diff
@@ -253,33 +266,61 @@ There is no command line variant of this stanza.
 
 .. _source-repository-package-fields:
 
-The :ref:`VCS fields<vcs-fields>` of ``source-repository-package`` are:
+The fields of ``source-repository-package`` are:
+
+.. _source-repository-package-type:
+
+.. cfg-field:: type: VCS tool
 
-..
-  data SourceRepositoryPackage f = SourceRepositoryPackage
```

Comment:
I was talking about maintainers not users, people writing and editing the docs and not people reading the docs.

Reply:
> Why are you getting rid of this comment?

To clarify, this is an [RST comment](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#comments).

Reply:
My opinion is that having it as a comment is too smart for its own good. No one expects to have such comments in the RST, so there's basically zero chance of Cabal devs finding it when they need it. On the other hand, it can confuse people editing the RST, and it just needlessly inflate the RST making it harder to work with.

I understand that this is a topic where reasonable people can disagree, and I'd certainly survive having this comment. But if it comes to a vote, I vote no for the above reasons.

Reply:
Agreed.

Reply:
Maybe we can keep these comments but move them to the VCS page.

```diff
$ diff cabal-package-description-file.rst cabal-project-description-file.rst --unified
 ..
-  data SourceRepo = SourceRepo
-    { repoKind :: RepoKind
-    , repoType :: Maybe RepoType
-    , repoLocation :: Maybe String
-    , repoModule :: Maybe String
-    , repoBranch :: Maybe String
-    , repoTag :: Maybe String
-    , repoSubdir :: Maybe FilePath
+  data SourceRepositoryPackage f = SourceRepositoryPackage
+    { srpType :: !RepoType
+    , srpLocation :: !String
+    , srpTag :: !(Maybe String)
+    , srpBranch :: !(Maybe String)
+    , srpSubdir :: !(f FilePath)
+    , srpCommand :: ![String]
     }
```

Reply:
Only if you also say in the first that it's a comment, I think. It already confused me, and that despite my having gone through the ReST documentation to learn about manual index entries.

Reply:
I'd pasted those two data declarations because they are close in their fields and annoyingly different in their names (why `SourceRepo` and not `SourceRepository` or why not `SourceRepo` and `SourceRepoPkg`?). Anyway that was my reason for adding these RST comments in the first place, to save a lookup into the code.

I'm using vscode with the [reStructured Text extension](https://marketplace.visualstudio.com/items?itemName=lextudio.restructuredtext) and it syntax highlights comments well enough:

![image](https://github.com/user-attachments/assets/4234fdb6-648a-437b-9018-5304c22815cf)

Reply:
I expect most devs use vim, though. I do use VSCodium, but only for code; docs I usually use vim for.

Reply:
If the comment is not useful to others (as I'd hoped) then I'm happy to see it go.